openwrt/files/root/.ssh/authorized_keys

```diff
@@ -1,4 +1 @@
-ssh-rsa AAAAB3NzaC1yc2EAAAADAQABAAABAQC4h99AAkDj+fq2K0hMv/SdgiXDaPQRc28ofuEGGW5ru+sjW51QGKsETXnTq5/Q5dO/AONKboM38qJ3BBimRJt3y1CsxMUpVE4A1Xq+PybROFbxuFogxfnEDJqonz+F/0wYC6SWkjctMv4b3qWbR0e3b3db4voy8XU/5sycqI9hRGdLkgjxP3AVmdwKyVZ21B0JITw03up9CNhy+e0tMs3Dw60vmRQMYrxO6Gg447ttMB7sAVHuCG85GSQjFSpee3J4efcQFgNO7V4BJvm7daIicFh5qzh+DRh/QDs1kHLOuf56C0h0HcKUY1V2fbmtBBOX4YzS38ywXY5RrVNDVnOZ dlang@localhost
-ssh-rsa AAAAB3NzaC1yc2EAAAADAQABAAACAQDrMp0YYH1epC8A3dci7gSPB/Prko4/ULY74kD/3kSrF/4FDfsfhTVo1on1SN42fdBbbUFc0rwoy1qhFETs+MmJIplmhU5d72nStR/jDjFbshKEtTFrg02uatVbq/xDCsdgq/uJ7UJaa5a8m5XEO/rqziPWipDe5qCj3V8pX/+oCBccnqtnTvY9WsxJioOwwSu45NbW7Bhg2hOHFCwj0qcGT7Gfa6EljAypOcxaGKH/aFsHM239OedXP9QAr7XiJ5px6EFvVSqkX3tqxpw6/7KnMrSrkC4OXacxIZyibvy0IdQm0fVljx876Sf5a7n9vZMvWhEEgxf3IH2b2zZzPtuR9HwoqTqnN37kBIeh0WpFwmkRLxgcueZdRv9PxZ2pIPrxRyMLNj4750n8GAmXzwaDv4fNUr6vMosj2umnxyWGFw48tq8Vn6b75MeadI2Tu07occJRKnYN+5W2OVWhwi17f2SZT9NdHBIJJYu1GuzG6e4EwHzTz+iqxRtjTWxrkDCcopKmwyc42seN/JdozzEaATAY0h13XhM1k26vNkafhCkWQcTaTpZpjTc1Voz0jPqQajjP4d1krHs8377lO1ZZ5ZADXFykgvGzflf93WREqo9MODlbXbkxpkvqiyBIZM/28IR/EsnHJhYvY5ucEAMovAyp1tu567QPZNJF6XRsKw==
-ecdsa-sha2-nistp256 AAAAE2VjZHNhLXNoYTItbmlzdHAyNTYAAAAIbmlzdHAyNTYAAABBBBjjcUJLTENGrV6K/nrPOswcBVMMuS4sLSs0UyTRw8wU87PDUzJz8Ht2SgHqeEQJdRm1+b6iLsx2uKOf+/pU8qE= root@kiev.delong.com
-ssh-rsa AAAAB3NzaC1yc2EAAAADAQABAAACAQDBP0XP83nOtxnsGJ7AG+FuT7dbAEO8BrIHlPbhCGhKjFKcq8g6LWa0htx3iwHJB2VfasIVSqWAC7SwX81XfSTwpvNSNVXvrIcMNdYOpEBHx5W94iVP2ooVVsGIeK/M+Eu9toiHebtVFT45hQ8VO5DwqbG6xPgEp1PmLsPhyBwP0LyC8Q4tA2ttftC6/fJa6fa1G54c7D/euIhZnyM/ovcpYfNYTlarX/yLJoYaOEk5ByYJPYnpQtm2GQX8CdT/U14nDtJ/VjzW8kP/R6+ZsmGpxBrWp2Mn03LjL4CzTSN3cbE2CAKKSjkIcaaoohRZS+JXnxm5+TXF/6QmN0MPU3SSThKWIzOhzfMrLqOu6hsw16x9lF1eJreUZr/WysWOswM4WNoffBv6mKvVsMR8Z8tgERR8+sIxUaoVOfChR9wRoD5lcNnVxrC6rcF+rBxeY4JiPS3zpyamwteyR0n2YspT9Dl0MlZb4cWml2aBnvYk0TmoCC3eGs+2/wssL4s+64TiFC8WkK7U7RW0eYODwPJQbzNGZBjcsVlKD4whs5cNGlgjkZJzTdFt49EwIKmFnSd9Ly/NY3zhByFS/MN3Mj9I2AqMRZ/i4iT72gATE9aLle1xViGxjK7bHmHT2H1yqQhJ5bMcorpDzSZyTkyBTDDnOgh8tM+tBG204kYywQW/RQ==
+{{ range (datasource "keys_dir") }}{{(file.Read (path.Join (getenv "KEYPATH") .))}}{{end}}
```

Comment:
I wanted to have gomplate do the authorized_keys file generation so there wouldn't be seperate steps to maintain in both the Makefile and build.sh. Unfortunately the only way I could get this to work is by using the environment variable KEYPATH to essentially pass the path into gomplate. If you try to interact with the datasource directly then the path is stripped and I couldn't hard code the location because it's different in build.sh and Makefile (one is ../../facts and the other is ../facts). 

I'd love to be able to omit it if there's a way, but this was still cleaner than other options I tried first.

Reply:
I follow you. I played around with this a little too and couldnt find a good way to deal with this either. Its weird how `keys_dir` basically gets evaluated into all files in that path but we can find the absolute path of the files themselves (even with other functions). Im good with this and its a straighforward solution 👍 

At some point, I think the easiest thing might be to unify the `Makefile` and `tests/unit/openwrt/test.sh` dir contexts when they execute `gomplate` to generate templates. Then we could hard code the path in `authorized_keys` or at least assume the relative path is the same to the pub keys. A nice side effect of this would be it would remove any fat fingers effect our tests since they would be making the exact same call. I can spin out a follow up for this so look into it further if you think thats a good idea.